pnpm dev not working with next.js source code

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #48-Ubuntu SMP Tue Jun 20 20:34:08 UTC 2023
    Binaries:
      Node: 18.17.1
      npm: 9.6.7
      Yarn: N/A
      pnpm: 8.6.11
    Relevant Packages:
      next: 13.4.20-canary.9
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: N/A
    Next.js Config:
      output: N/A
```


### Which area(s) of Next.js are affected? (leave empty if unsure)

Package manager (npm, pnpm, Yarn), Turbopack (--turbo)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/vercel/next.js

### To Reproduce


1. Install the [GitHub CLI](https://github.com/cli/cli#installation).
1. Clone the Next.js repository (download only recent commits for faster clone):
   ```
   gh repo clone vercel/next.js -- --depth=3000 --branch canary --single-branch
   ```
1. Enable pnpm:
   ```
   corepack enable pnpm
   ```
1. Install the dependencies with:
   ```
   pnpm install
   ```
1. Start developing and watch for code changes:
   ```
   pnpm dev
   ```

### Describe the Bug

I was trying to start contributing to nextjs open source, and the official code doesn't seem to work!

During pnpm install

```
...
 WARN  Failed to create bin at /workspaces/next.js/bench/nested-deps-app-router/node_modules/.bin/next. ENOENT: no such file or directory, open '/workspaces/next.js/packages/next/dist/bin/next'
 WARN  Failed to create bin at /workspaces/next.js/bench/next-minimal-server/node_modules/.bin/next. ENOENT: no such file or directory, open '/workspaces/next.js/packages/next/dist/bin/next'
 WARN  Failed to create bin at /workspaces/next.js/bench/nested-deps/node_modules/.bin/next. ENOENT: no such file or directory, open '/workspaces/next.js/packages/next/dist/bin/next'
 WARN  Failed to create bin at /workspaces/next.js/bench/app-router-server/node_modules/.bin/next. ENOENT: no such file or directory, open '/workspaces/next.js/packages/next/dist/bin/next'
 WARN  Failed to create bin at /workspaces/next.js/packages/next-plugin-storybook/node_modules/.bin/next. ENOENT: no such file or directory, open '/workspaces/next.js/packages/next/dist/bin/next'
 WARN  5 other warnings
...
```

During pnpm dev

```
...
7:58:04 AM - Starting compilation in watch mode...
@next/third-parties:dev: 
bench-nested-deps-app-router:dev: node:internal/modules/cjs/loader:1077
bench-nested-deps-app-router:dev:   const err = new Error(message);
bench-nested-deps-app-router:dev:               ^
bench-nested-deps-app-router:dev: 
bench-nested-deps-app-router:dev: Error: Cannot find module 'webpack/lib/javascript/BasicEvaluatedExpression'
bench-nested-deps-app-router:dev: Require stack:
bench-nested-deps-app-router:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/compiled/webpack/webpack.js
bench-nested-deps-app-router:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config-utils.js
bench-nested-deps-app-router:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config.js
bench-nested-deps-app-router:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/cli/next-dev.js
bench-nested-deps-app-router:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/lib/commands.js
bench-nested-deps-app-router:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/bin/next
bench-nested-deps-app-router:dev:     at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
bench-nested-deps-app-router:dev:     at Module._load (node:internal/modules/cjs/loader:922:27)
bench-nested-deps-app-router:dev:     at Module.require (node:internal/modules/cjs/loader:1143:19)
bench-nested-deps-app-router:dev:     at require (node:internal/modules/cjs/helpers:121:18)
bench-nested-deps-app-router:dev:     at exports.init (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/compiled/webpack/webpack.js:9:33)
bench-nested-deps-app-router:dev:     at loadWebpackHook (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config-utils.js:18:5)
bench-nested-deps-app-router:dev:     at loadConfig (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config.js:535:46)
bench-nested-deps-app-router:dev:     at nextDev (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/cli/next-dev.js:281:40) {
bench-nested-deps-app-router:dev:   code: 'MODULE_NOT_FOUND',
bench-nested-deps-app-router:dev:   requireStack: [
bench-nested-deps-app-router:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/compiled/webpack/webpack.js',
bench-nested-deps-app-router:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config-utils.js',
bench-nested-deps-app-router:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config.js',
bench-nested-deps-app-router:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/cli/next-dev.js',
bench-nested-deps-app-router:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/lib/commands.js',
bench-nested-deps-app-router:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/bin/next'
bench-nested-deps-app-router:dev:   ]
bench-nested-deps-app-router:dev: }
bench-nested-deps-app-router:dev: 
bench-nested-deps-app-router:dev: Node.js v18.17.1
bench-nested-deps-app-router:dev:  ELIFECYCLE  Command failed with exit code 1.
bench-nested-deps:dev: node:internal/modules/cjs/loader:1077
bench-nested-deps:dev:   const err = new Error(message);
bench-nested-deps:dev:               ^
bench-nested-deps:dev: 
bench-nested-deps:dev: Error: Cannot find module 'webpack/lib/javascript/BasicEvaluatedExpression'
bench-nested-deps:dev: Require stack:
bench-nested-deps:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/compiled/webpack/webpack.js
bench-nested-deps:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config-utils.js
bench-nested-deps:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config.js
bench-nested-deps:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/cli/next-dev.js
bench-nested-deps:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/lib/commands.js
bench-nested-deps:dev: - /usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/bin/next
bench-nested-deps:dev:     at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
bench-nested-deps:dev:     at Module._load (node:internal/modules/cjs/loader:922:27)
bench-nested-deps:dev:     at Module.require (node:internal/modules/cjs/loader:1143:19)
bench-nested-deps:dev:     at require (node:internal/modules/cjs/helpers:121:18)
bench-nested-deps:dev:     at exports.init (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/compiled/webpack/webpack.js:9:33)
bench-nested-deps:dev:     at loadWebpackHook (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config-utils.js:18:5)
bench-nested-deps:dev:     at loadConfig (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config.js:535:46)
bench-nested-deps:dev:     at nextDev (/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/cli/next-dev.js:281:40) {
bench-nested-deps:dev:   code: 'MODULE_NOT_FOUND',
bench-nested-deps:dev:   requireStack: [
bench-nested-deps:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/compiled/webpack/webpack.js',
bench-nested-deps:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config-utils.js',
bench-nested-deps:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/server/config.js',
bench-nested-deps:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/cli/next-dev.js',
bench-nested-deps:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/lib/commands.js',
bench-nested-deps:dev:     '/usr/local/share/nvm/versions/node/v18.17.1/lib/node_modules/next/dist/bin/next'
bench-nested-deps:dev:   ]
bench-nested-deps:dev: }
bench-nested-deps:dev: 
bench-nested-deps:dev: Node.js v18.17.1
bench-nested-deps:dev:  ELIFECYCLE  Command failed with exit code 1.
bench-nested-deps-app-router:dev: ERROR: command finished with error: command (/workspaces/next.js/bench/nested-deps-app-router) pnpm run dev exited (1)
create-next-app:dev:  ELIFECYCLE  Command failed.
next:dev:  ELIFECYCLE  Command failed.
@next/font:dev:  ELIFECYCLE  Command failed.
@next/third-parties:dev:  ELIFECYCLE  Command failed.
@next/polyfill-module:dev:  ELIFECYCLE  Command failed.
@next/react-refresh-utils:dev:  ELIFECYCLE  Command failed.
@next/env:dev:  ELIFECYCLE  Command failed.
@next/codemod:dev:  ELIFECYCLE  Command failed.
@next/react-dev-overlay:dev:  ELIFECYCLE  Command failed.
@next/polyfill-nomodule:dev:  ELIFECYCLE  Command failed.
bench-nested-deps-app-router#dev: command (/workspaces/next.js/bench/nested-deps-app-router) pnpm run dev exited (1)

 Tasks:    0 successful, 12 total
Cached:    0 cached, 12 total
  Time:    8.365s 
Failed:    bench-nested-deps-app-router#dev

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
...
```


### Expected Behavior

Start developing and watch for code changes:

### Which browser are you using? (if relevant)

Irrelevant

### How are you deploying your application? (if relevant)

pnpm dev

--- a/bench/nested-deps-app-router/package.json
+++ b/bench/nested-deps-app-router/package.json
@@ -2,10 +2,10 @@
   "name": "bench-nested-deps-app-router",
   "scripts": {
     "prepare-bench": "rimraf components && fuzzponent -d 2 -s 206 -o components",
-    "dev": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next dev",
+    "dev-application": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next dev",
     "build-application": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next build",
     "start": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next start",
-    "dev-nocache": "rimraf .next && pnpm dev",
+    "dev-nocache": "rimraf .next && pnpm dev-application",
     "dev-cpuprofile-nocache": "rimraf .next && cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 node --cpu-prof ../../node_modules/next/dist/bin/next",
     "build-nocache": "rimraf .next && pnpm build-application"
   },

--- a/bench/nested-deps/package.json
+++ b/bench/nested-deps/package.json
@@ -2,10 +2,10 @@
   "name": "bench-nested-deps",
   "scripts": {
     "prepare-bench": "rimraf components && fuzzponent -d 2 -s 206 -o components",
-    "dev": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next dev",
+    "dev-application": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next dev",
     "build-application": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next build",
     "start": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 next start",
-    "dev-nocache": "rimraf .next && pnpm dev",
+    "dev-nocache": "rimraf .next && pnpm dev-application",
     "dev-cpuprofile-nocache": "rimraf .next && cross-env NEXT_PRIVATE_LOCAL_WEBPACK=1 node --cpu-prof ../../node_modules/next/dist/bin/next",
     "build-nocache": "rimraf .next && pnpm build-application"
   },

